meta.jq

```diff
@@ -53,10 +53,24 @@ def pull_command:
 	end
 ;
 # input: "build" object (with "buildId" top level key)
+# output: string "giturl" ("https://github.com/docker-library/golang#commit:directory), used for "docker build giturl"
+def git_build_url:
+	.source.entry.GitRepo as $gitRepo
+	| if ($gitRepo | test("^https?://github.com/"))
+		and ($gitRepo | endswith(".git") | not) then
+			# without ".git" in the url "docker build url" fails and tries to build the html repo page as a Dockerfile
+			# https://github.com/moby/moby/blob/4f9c865eddc09da0c0cb9fe08c76b81b804093f5/builder/remotecontext/urlutil/urlutil.go#L15
```

Comment:
(included in the better later suggestion)

```suggestion
			# without ".git" in the url "docker buildx build url" fails and tries to build the html repo page as a Dockerfile
			# https://github.com/moby/buildkit/blob/0e1e36ba9eb8142968b2c5cfa2f12549bf9246d9/util/gitutil/git_ref.go#L81-L87
```

Reply:
That regular expression is `".git(?:#.+)?$"`, which allows for anchors while `($gitRepo | endswith(".git") | not)` does not. Is this okay?

Reply:
It appears that BuildKit code parses the URL and matches specifically on the Path portion of the URL, which does not include any query or fragment portions of the URL.

Reply:
Yeah, I don't think `git clone` on a URL with a query parameter would actually work correctly, but maybe there's some esoteric server somewhere?  I'm not worried about the regex being `.git` because that's just a bug (a 6 year old bug, to be fair, so could be considered "load bearing", but BuildKit implemented the same check more correctly and nobody's noticed, so I don't think anyone's put real load on it)

Reply:
(In other words, any `GitRepo:` with an anchor in it would be invalid already)